### DISCUSSION TITLE
User_Setup.h HY32D (ILI9341 maybe)

### DISCUSSION THREAD

**Discussion by JayNaire** [Bodmer/TFT_eSPI#2489] [Compatible support libraries]
linux->vscode->platformio:
platform = espressif32
board = esp32doit-devkit-v1
framework = arduino

I have TFT_eSPI working really well with an ESP32 and some 2.8" ILI9341+XPT2046 display. All in SPI mode.

I'd like to substitute another display labelled "HY32D". 16bit Parallel. Touch chip is XPT2046. TFT chip is unknown (might be ILI9341 - but it seems HY32D came in many flavours). Looks like [this.](https://www.aliexpress.com/item/1787431150.html)

I tried building with defs from User_Setup.h (TFT_PARALLEL_8_BIT, etc.)
Computer said: _"tft.setTouch: Touch functions not supported in 8/16 bit parallel mode..."_
Which is rather definitive.

Please tick a box.
[_] Bad luck, you've been told: no parallel with touch. Use your HY32D for a non-touch application.

[_] Well you could set the ILI9341 for MISO/MOSI serial operation. Best of luck finding which pins to connect without a schematic of your specific HY32D. Always assuming you really do have an  ILI9341 in the first place. And you're happy hacking SPI to arbitrate between tft and touch.

[_] Tch! Loads of people have ESP32 + HY32D touch displays. What you do is this...

Many thanks for any thoughts.

**Comment by Bodmer** [comment]
16 bit parallel displays are not supported with ESP32. The RP2040 only has 16 bit parallel support.

**Reply by JayNaire** [reply]
Fair enough. Thanks for that.

**Reply by Bodmer** [reply]
It looks like some of the displays in your web link have board configuration options to select different interfaces.
